crates/chain/src/local_chain.rs

```diff
@@ -0,0 +1,184 @@
+use core::{convert::Infallible, ops::Deref};
+
+use alloc::{
+    collections::{BTreeMap, BTreeSet},
+    vec::Vec,
+};
+use bitcoin::BlockHash;
+
+use crate::{BlockId, ChainOracle};
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct LocalChain {
+    blocks: BTreeMap<u32, BlockHash>,
+}
```

Comment:
Is it intended for `LocalChain` to replace `SparseChain`?

Reply:
Yessir!